homeassistant/components/homee/cover.py

```diff
@@ -0,0 +1,262 @@
+
+import logging
+from typing import Any, cast
+
+from pyHomee.const import AttributeType, NodeProfile
+from pyHomee.model import HomeeNode
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeeConfigEntry, helpers
+from .entity import HomeeNodeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+OPEN_CLOSE_ATTRIBUTES = [
+    AttributeType.OPEN_CLOSE,
+    AttributeType.SLAT_ROTATION_IMPULSE,
+    AttributeType.UP_DOWN,
+]
+POSITION_ATTRIBUTES = [AttributeType.POSITION, AttributeType.SHUTTER_SLAT_POSITION]
+
+
+def get_cover_features(
+    node: HomeeNodeEntity,
+) -> tuple[CoverEntityFeature, AttributeType]:
+    """Determine the supported cover features of a homee node based on the available attributes."""
+    features = CoverEntityFeature(0)
+    open_close = None
+
+    # We assume, that no device has UP_DOWN and OPEN_CLOSE, but only one of them.
+    if node.has_attribute(AttributeType.UP_DOWN) or node.has_attribute(
+        AttributeType.OPEN_CLOSE
+    ):
+        if node.has_attribute(AttributeType.UP_DOWN):
+            open_close = AttributeType.UP_DOWN
+        else:
+            open_close = AttributeType.OPEN_CLOSE
+
+        if node.get_attribute(open_close).editable:
+            features |= (
+                CoverEntityFeature.OPEN
+                | CoverEntityFeature.CLOSE
+                | CoverEntityFeature.STOP
+            )
+
+    # Check for up/down position settable.
+    if node.has_attribute(AttributeType.POSITION):
+        if node.get_attribute(AttributeType.POSITION).editable:
+            features |= CoverEntityFeature.SET_POSITION
+
+    if node.has_attribute(AttributeType.SLAT_ROTATION_IMPULSE):
+        features |= CoverEntityFeature.OPEN_TILT | CoverEntityFeature.CLOSE_TILT
+
+    if node.has_attribute(AttributeType.SHUTTER_SLAT_POSITION):
+        features |= CoverEntityFeature.SET_TILT_POSITION
+
+    return features, open_close
+
+
+def get_device_class(node: HomeeNode) -> CoverDeviceClass | None:
+    """Determine the device class a homee node based on the node profile."""
+    if node.profile == NodeProfile.GARAGE_DOOR_OPERATOR:
+        return CoverDeviceClass.GARAGE
+
+    if node.profile == NodeProfile.SHUTTER_POSITION_SWITCH:
+        return CoverDeviceClass.SHUTTER
+
+    return None
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HomeeConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Add the homee platform for the cover integration."""
+
+    devices: list[HomeeCover] = []
+    nodes = helpers.get_imported_nodes(config_entry)
+    devices.extend(
+        HomeeCover(node, config_entry) for node in nodes if is_cover_node(node)
+    )
+
+    if devices:
+        async_add_devices(devices)
```

Comment:
```suggestion
    async_add_devices(
        HomeeCover(node, config_entry) for node in config_entry.runtime_data.nodes if is_cover_node(node)
    )
```
This would be enough here

Reply:
Nice. 🙂